x-pack/plugin/esql/src/main/java/generated/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvAvgDoubleEvaluator.java

```diff
@@ -34,55 +34,51 @@ public String name() {
    * Evaluate blocks containing at least one multivalued field.
    */
   @Override
-  public Block.Ref evalNullable(Block.Ref ref) {
-    try (ref) {
-      DoubleBlock v = (DoubleBlock) ref.block();
-      int positionCount = v.getPositionCount();
-      try (DoubleBlock.Builder builder = driverContext.blockFactory().newDoubleBlockBuilder(positionCount)) {
-        CompensatedSum work = new CompensatedSum();
-        for (int p = 0; p < positionCount; p++) {
-          int valueCount = v.getValueCount(p);
-          if (valueCount == 0) {
-            builder.appendNull();
-            continue;
-          }
-          int first = v.getFirstValueIndex(p);
-          int end = first + valueCount;
-          for (int i = first; i < end; i++) {
-            double value = v.getDouble(i);
-            MvAvg.process(work, value);
-          }
-          double result = MvAvg.finish(work, valueCount);
-          builder.appendDouble(result);
+  public Block evalNullable(Block fieldVal) {
+    DoubleBlock v = (DoubleBlock) fieldVal;
```

Comment:
question: I guess `fieldVal` was never a floating `Block.Ref` when passed in here? We don't seem to need to close this block, so it probably is always part of a page and released via `Page.releaseBlocks`, right?

Reply:
That's correct. We should not release the input block.